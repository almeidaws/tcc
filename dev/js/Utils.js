/** 
 * This class is used as namespace to contains generic method used every
 * where in the application.
 */
class Utils {

    /**
     * Adds a cookie to the browser.
     * @param {string} cname cookie's name/key.
     * @param {string} cvalue cookie's value.
     * @param {number} exdays cookie's duration.
     */
    static setCookie(cname, cvalue, exdays) {
      var d = new Date();
      d.setTime(d.getTime() + (exdays*24*60*60*1000));
      var expires = "expires="+ d.toUTCString();
      document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
    }

    /**
     * Deletes a cookie from user's browser.
     * @param {string} cookie's name.
     */
    static getCookie(cname) {
      var name = cname + "=";
      var decodedCookie = decodeURIComponent(document.cookie);
      var ca = decodedCookie.split(';');
      for(var i = 0; i <ca.length; i++) {
        var c = ca[i];
        while (c.charAt(0) == ' ') {
          c = c.substring(1);
        }
        if (c.indexOf(name) == 0) {
          return c.substring(name.length, c.length);
        }
      }
      return "";
    }
    
    /**
     * Deletes all cookies from user's broswer. It was
     * created to be used on automatic tests.
     */
    static deleteAllCookies() {
        var cookies = document.cookie.split(";");

        for (var i = 0; i < cookies.length; i++) {
            var cookie = cookies[i];
            var eqPos = cookie.indexOf("=");
            var name = eqPos > -1 ? cookie.substr(0, eqPos) : cookie;
            document.cookie = name + "=;expires=Thu, 01 Jan 1970 00:00:00 GMT";
        }
    }
}

export default Utils;
