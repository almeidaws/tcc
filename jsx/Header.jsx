const registerButton = <a href="javascript:;" className="ms_btn login_btn" data-toggle="modal" data-target="#myModal"><span>Register</span></a>;
const loginButton = <a href="javascript:;" className="ms_btn login_btn" data-toggle="modal" data-target="#myModal1"><span>login</span></a>;
const searchButton = (
    <div>
        <input type="text" className="form-control" placeholder="Search Music Here.." />
        <span className="search_icon">
            <img src="images/svg/search.svg" alt="" />
        </span>
    </div>
);

const unloggedMenu = <div className="ms_top_btn">{registerButton}{loginButton}</div>;

const LoggedMenu = props => (
    <div className="ms_top_btn">
        <a href="javascript:;" className="ms_admin_name">Hello {props.name} <span className="ms_pro_name">{props.name.charAt(0)}</span>													
        </a>
        <ul className="pro_dropdown_menu">
            <li><a href="profile.html">Profile</a></li>
            <li><a href="manage_acc.html" target="_blank">Pricing Plan</a></li>
            <li><a href="blog.html" target="_blank">Blog</a></li>
            <li><a href="">Setting</a></li>
            <li><a href="">Logout</a></li>
        </ul>
    </div>
);

class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = { name: getCookie('name') };
    }
    render() {
        return (
            <div className="ms_header">
                <div className="ms_top_left">
                    <div className="ms_top_search">
                        {searchButton}
                    </div>
                </div>
                <div className="ms_top_right">
                    {this.state.name ? LoggedMenu({ name: this.state.name }) : unloggedMenu}
                </div>
            </div>
       );
   }
};

ReactDOM.render(<Header />, document.getElementById('header'));
