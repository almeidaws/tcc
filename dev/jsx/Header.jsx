import React from 'react';
import ReactDOM from 'react-dom';
import Requisition from './../js/Requisition.js';
const R = Requisition;
import Utils from './../js/Utils.js';
const U = Utils;

const registerButton = <a href="javascript:;" className="ms_btn login_btn" data-toggle="modal" data-target="#myModal"><span>Register</span></a>;
const loginButton = <a href="javascript:;" className="ms_btn login_btn" data-toggle="modal" data-target="#myModal1"><span>login</span></a>;
const searchButton = (
    <div>
        <input type="text" className="form-control" placeholder="Search Music Here.." />
        <span className="search_icon">
            <img src="images/svg/search.svg" alt="" />
        </span>
    </div>
);

const unloggedMenu = <div className="ms_top_btn">{registerButton}{loginButton}</div>;

/**
 * Menu presented on website's header when the user is currently logged. It currently only
 * presents the user's name provided through props.
 *
 * Property name: users'name. Example: 'Carlos'.
 */
const LoggedMenu = props => (
    <div className="ms_top_btn">
        <a href="javascript:;" className="ms_admin_name">Hello {props.name} <span className="ms_pro_name">{props.name.charAt(0)}</span>													
        </a>
        <ul className="pro_dropdown_menu">
            <li><a href="profile.html">Profile</a></li>
            <li><a href="manage_acc.html" target="_blank">Pricing Plan</a></li>
            <li><a href="blog.html" target="_blank">Blog</a></li>
            <li><a href="">Setting</a></li>
            <li><a href="">Logout</a></li>
        </ul>
    </div>
);

/**
 * Renders the website's header presented on top of each page.
 * This component conditionally present buttons for registering/logging
 * or user's information if the user is logged or not.
 *
 * To know if the user is logged this components checks the if there's 
 * a cookie named 'name' that contains user's name.
 */
class Header extends React.Component {
    constructor(props) {
        super(props);
        this.state = { name: U.getCookie('name') };
    }
    render() {
        return (
            <div className="ms_header">
                <div className="ms_top_left">
                    <div className="ms_top_search">
                        {searchButton}
                    </div>
                </div>
                <div className="ms_top_right">
                    {this.state.name ? LoggedMenu({ name: this.state.name }) : unloggedMenu}
                </div>
            </div>
       );
   }
};

export default Header;
