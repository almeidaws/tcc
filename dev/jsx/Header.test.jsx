import React from 'react';
import Header from './Header.jsx';
import renderer from 'react-test-renderer';
import Utils from './../js/Utils.js';
const U = Utils;

beforeEach(() => { U.deleteAllCookies });

it('<Header /> should render register/login buttons', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('<Header /> should render user\'s profile menus and name', () => {
    U.setCookie('name', 'Carlos', 1);
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
