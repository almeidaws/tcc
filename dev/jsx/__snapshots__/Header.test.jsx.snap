// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render register/login buttons 1`] = `
<div
  className="ms_header"
>
  <div
    className="ms_top_left"
  >
    <div
      className="ms_top_search"
    >
      <div>
        <input
          className="form-control"
          placeholder="Search Music Here.."
          type="text"
        />
        <span
          className="search_icon"
        >
          <img
            alt=""
            src="images/svg/search.svg"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ms_top_right"
  >
    <div
      className="ms_top_btn"
    >
      <a
        className="ms_btn login_btn"
        data-target="#myModal"
        data-toggle="modal"
        href="javascript:;"
      >
        <span>
          Register
        </span>
      </a>
      <a
        className="ms_btn login_btn"
        data-target="#myModal1"
        data-toggle="modal"
        href="javascript:;"
      >
        <span>
          login
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`<Header /> should render user's profile menus and name 1`] = `
<div
  className="ms_header"
>
  <div
    className="ms_top_left"
  >
    <div
      className="ms_top_search"
    >
      <div>
        <input
          className="form-control"
          placeholder="Search Music Here.."
          type="text"
        />
        <span
          className="search_icon"
        >
          <img
            alt=""
            src="images/svg/search.svg"
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ms_top_right"
  >
    <div
      className="ms_top_btn"
    >
      <a
        className="ms_admin_name"
        href="javascript:;"
      >
        Hello 
        Carlos
         
        <span
          className="ms_pro_name"
        >
          C
        </span>
      </a>
      <ul
        className="pro_dropdown_menu"
      >
        <li>
          <a
            href="profile.html"
          >
            Profile
          </a>
        </li>
        <li>
          <a
            href="manage_acc.html"
            target="_blank"
          >
            Pricing Plan
          </a>
        </li>
        <li>
          <a
            href="blog.html"
            target="_blank"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            href=""
          >
            Setting
          </a>
        </li>
        <li>
          <a
            href=""
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
